sway/desktop/idle_inhibit_v1.c

```diff
@@ -29,28 +35,93 @@ void handle_idle_inhibitor_v1(struct wl_listener *listener, void *data) {
 	}
 
 	inhibitor->manager = manager;
+	inhibitor->mode = INHIBIT_IDLE_APPLICATION;
 	inhibitor->view = view_from_wlr_surface(wlr_inhibitor->surface);
 	wl_list_insert(&manager->inhibitors, &inhibitor->link);
 
-
 	inhibitor->destroy.notify = handle_destroy;
 	wl_signal_add(&wlr_inhibitor->events.destroy, &inhibitor->destroy);
 
 	idle_inhibit_v1_check_active(manager);
 }
 
+void idle_inhibit_v1_user_inhibitor_register(struct sway_view *view,
+		enum sway_idle_inhibit_mode mode) {
+	struct sway_idle_inhibitor_v1 *inhibitor =
+		calloc(1, sizeof(struct sway_idle_inhibitor_v1));
+	if (!inhibitor) {
+		return;
+	}
+
+	inhibitor->manager = server.idle_inhibit_manager_v1;
+	inhibitor->mode = mode;
+	inhibitor->view = view;
+	wl_list_insert(&inhibitor->manager->inhibitors, &inhibitor->link);
+
+	inhibitor->destroy.notify = handle_destroy;
+	wl_signal_add(&view->events.unmap, &inhibitor->destroy);
+
+	idle_inhibit_v1_check_active(inhibitor->manager);
+}
+
+struct sway_idle_inhibitor_v1 *idle_inhibit_v1_user_inhibitor_for_view(
+		struct sway_view *view) {
+	struct sway_idle_inhibitor_v1 *inhibitor;
+	wl_list_for_each(inhibitor, &server.idle_inhibit_manager_v1->inhibitors,
+			link) {
+		if (inhibitor->view == view &&
+				inhibitor->mode != INHIBIT_IDLE_APPLICATION) {
+			return inhibitor;
+		}
+	}
+	return NULL;
+}
+
+void idle_inhibit_v1_user_inhibitor_destroy(
```

Comment:
nit: the `idle_inhibit_v1_*` namespace belongs to wlroots

Reply:
Should I rename the existing `idle_inhibit_v1_check_active` as well?

Reply:
Hm, probably.

Reply:
Changed